tools/gui/qtcodegen.py

```diff
@@ -0,0 +1,335 @@
+        """
+        try:
+            self.plz_impfzentrum = self.kontaktdaten["plz_impfzentren"][0]
+            self.mail = self.kontaktdaten["kontakt"]["notificationReceiver"]
+            self.telefonnummer = self.kontaktdaten["kontakt"]["phone"]
+        except KeyError as error:
+            self.signalShowDlg.emit("MISSING_KONTAKT","")
+            self.stop()
+           
+        if self.stopped is True:
+            return False
+            
+        its = ImpfterminService([], {}, self.ROOT_PATH)
+        
+        # send signal for GUI
+        self.sendSignalAndWait("signalShowInput","GEBURTSDATUM")
+
+        #stop requested in the meanwhile?
+        if self.stopped is True:
+            return False
+            
+        # code anfordern
+        try:
+            token, cookies = its.code_anfordern(self.mail, self.telefonnummer,  self.plz_impfzentrum, self.geburtsdatum)
+        except RuntimeError as exc:
+            print(
+                f"\nDie Code-Generierung war leider nicht erfolgreich:\n{str(exc)}")
+            self.signalShowDlg.emit("CRITICAL_CLOSE",f"\nDie Code-Generierung war leider nicht erfolgreich:\n{str(exc)}")
+            while True and self.stopped is False:
+                QtCore.QThread.msleep(100)
+            return False
+
+        # code bestätigen            
+        # 3 Versuche für die SMS-Code-Eingabe
+        self.sendSignalAndWait("signalShowInput","SMSCODE")
+            
+        #stop requested in the meanwhile?
+        if self.stopped is True:
+            return False
+            
+        if its.code_bestaetigen(token, cookies, self.sms_pin, self.plz_impfzentrum):
+            self.sendSignalAndWait("signalShowInput","SMSCODE_OK")
+            return True
+
+        print("\nSMS-Code ungültig")
+        print("Die Code-Generierung war leider nicht erfolgreich.")
+
+        self.signalShowDlg.emit("CRITICAL_CLOSE",f"SMS-Code ungültig.\n\nDie Code-Generierung war leider nicht erfolgreich")
+        return False
+
+
+
+
+class QtCodeGen(QtWidgets.QDialog):
+    # Folgende Widgets stehen zur Verfügung:
+    def __init__(self, kontaktdaten: dict, ROOT_PATH: str, parent = None):
+        super().__init__(parent)
+        uic.loadUi(os.path.join(PATH, "ui_qtcodegen.ui"), self)
+        self.setupUi(self)
+        
+        self.parent = parent
+        self._hardClose = False
+        
+        # Attribute erstellen
+        self.kontaktdaten = kontaktdaten
+        self.ROOT_PATH = ROOT_PATH
+
+        # std.out & error auf das Textfeld umleiten
+        sys.stdout = EigenerStream(text_schreiben=self.update_ausgabe)
+        sys.stderr = EigenerStream(text_schreiben=self.update_ausgabe)
+
+        # Entsprechend Konfigurieren
+        self.setup_thread()
+        self.thread.start()
+        
+        # show gui
+        self.show()
+  
+    def __del__(self):
+        print("QtCodeGen destruct")
+        
+    def setupUi(self, QtCodeGen):
```

Comment:
Übergebe Param da self.ROOT_PATH noch nicht verfügbar ist
```suggestion
    def setupUi(self, QtCodeGen, ROOT_PATH):
```